src/Models/Repository/IRepository.cs

```diff
@@ -0,0 +1,19 @@
+﻿using Microsoft.Extensions.Hosting;
+using System;
+
+namespace Phytime.Models
+{
+    public interface IRepository<T, P> : IDisposable
```

Comment:
JFYI, generic repsitory is kinda anti-pattern, but I am OK with that now. You may read more on the internet